File: types/dish.ts
/**
 * Категории блюд
 */
export type DishCategory =
  | 'Drinks'
  | 'Salads'
  | 'Soups'
  | 'Mains'
  | 'Desserts'
  | 'ColdSnacks'
  | 'HotSnacks';

/**
 * Блюдо в меню
 */
export interface Dish {
  id: number;
  name: string;
  composition: string; // Состав
  weight: string; // Формат: 100/20/50
  price: number;
  category: DishCategory;
  cookingTime: number; // Время приготовления в минутах
  tags: string[]; // Теги: веган, острое, халяль, кошер и т.д.
  imageUrl?: string;
}

/**
 * Создание/обновление блюда
 */
export interface DishInput {
  name: string;
  composition: string;
  weight: string;
  price: number;
  category: DishCategory;
  cookingTime: number;
  tags: string[];
  imageUrl?: string;
}

/**
 * Меню с категориями
 */
export interface MenuData {
  categories: DishCategory[];
  dishes: Dish[];
}

File: Prog_korp_seti-master/backend/Restaurant.Domain/Entities/Dish.cs
namespace Restaurant.Domain.Entities;

public class Dish
{
    public int Id { get; set; }
    public string Name { get; set; } = default!;
    public string Composition { get; set; } = default!;
    public string Weight { get; set; } = default!;
    public decimal Price { get; set; }
    public int CategoryId { get; set; } // Foreign key to DishCategory
    public int CookingTime { get; set; }
    public bool IsAvailable { get; set; } = true;
    public bool IsDeleted { get; set; } = false;
    public string? ImageUrl { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation properties
    public DishCategory? Category { get; set; }
    public ICollection<DishType>? Types { get; set; } // Many-to-many
    public ICollection<OrderItem>? OrderItems { get; set; }
}

